.github/workflows/lint-test.yml

```diff
@@ -27,6 +27,12 @@ jobs:
           node-version-file: 'package.json'
           cache: 'pnpm'
 
+      - name: Cache Cypress binary 📌
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/Cypress
+          key: cypress-${{ runner.os }}-${{ hashFiles('pnpm-lock.yaml') }}
+
```

Comment:
When the `pnpm-lock.yaml` changes and we push code to the repo, if the `lint` job succeeds but not the `e2e`, we end up with a cache for the pnpm store but not the Cypress binaries. When that happens, on subsequent pushes, the `e2e` job always fails because it can never find the binaries, and the binaries never get cached. Only way to resolve is to change `pnpm-lock.yaml`, push and hope that the `e2e` passes.

To resolve this, I now cache the Cypress binaries in every job. This is not ideal. I've opened an issue in `actions/setup-node` to solve this more elegantly: https://github.com/actions/setup-node/issues/638

Reply:
Fine for now, thanks for investigating other ways :+1: